// [[Reason: Function declarations are not fixed]] ////

//// [tests/cases/conformance/salsa/typeFromPropertyAssignment29.ts] ////

===================================================================
--- TSC	declarations
+++ DTE	declarations
@@ -2,10 +2,10 @@
 
 //// [typeFromPropertyAssignment29.d.ts]
 declare function ExpandoDecl(n: number): string;
 declare namespace ExpandoDecl {
-    var prop: number;
-    var m: (n: number) => number;
+    var prop: invalid;
+    var m: invalid;
 }
 declare var n: number;
 declare const ExpandoExpr: {
     (n: number): string;
@@ -28,13 +28,13 @@
     (m: number): number;
     total: number;
 };
 declare namespace ExpandoNested {
-    var also: number;
+    var also: invalid;
 }
 declare function ExpandoMerge(n: number): number;
 declare namespace ExpandoMerge {
-    var p1: number;
+    var p1: invalid;
 }
 declare namespace ExpandoMerge {
     var p2: number;
 }
@@ -44,9 +44,9 @@
 declare var n: number;
 declare namespace Ns {
     function ExpandoNamespace(): void;
     namespace ExpandoNamespace {
-        var p6: number;
+        var p6: invalid;
     }
     export function foo(): typeof ExpandoNamespace;
     export {};
 }
@@ -55,37 +55,43 @@
 declare class ExpandoClass {
     n: number;
 }
 declare var n: number;
-declare var ExpandoExpr3: {
-    new (): {
-        n: number;
-    };
-};
+declare var ExpandoExpr3: invalid;
 declare var n: number;
 //# sourceMappingURL=typeFromPropertyAssignment29.d.ts.map
 /// [Errors] ////
 
+typeFromPropertyAssignment29.ts(4,1): error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+typeFromPropertyAssignment29.ts(5,1): error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+typeFromPropertyAssignment29.ts(51,1): error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+typeFromPropertyAssignment29.ts(56,1): error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
+typeFromPropertyAssignment29.ts(67,5): error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
 typeFromPropertyAssignment29.ts(77,14): error TS2339: Property 'prop' does not exist on type '(n: number) => string'.
 typeFromPropertyAssignment29.ts(78,14): error TS2339: Property 'm' does not exist on type '(n: number) => string'.
 typeFromPropertyAssignment29.ts(81,22): error TS2339: Property 'prop' does not exist on type '(n: number) => string'.
 typeFromPropertyAssignment29.ts(81,42): error TS2339: Property 'm' does not exist on type '(n: number) => string'.
 typeFromPropertyAssignment29.ts(87,14): error TS2339: Property 'prop' does not exist on type 'typeof ExpandoClass'.
 typeFromPropertyAssignment29.ts(88,14): error TS2339: Property 'm' does not exist on type 'typeof ExpandoClass'.
 typeFromPropertyAssignment29.ts(91,22): error TS2339: Property 'prop' does not exist on type 'typeof ExpandoClass'.
 typeFromPropertyAssignment29.ts(91,42): error TS2339: Property 'm' does not exist on type 'typeof ExpandoClass'.
+typeFromPropertyAssignment29.ts(94,20): error TS9022: Inference from class expressions is not supported with --isolatedDeclarations.
 typeFromPropertyAssignment29.ts(97,14): error TS2339: Property 'prop' does not exist on type 'typeof ExpandoExpr3'.
 typeFromPropertyAssignment29.ts(98,14): error TS2339: Property 'm' does not exist on type 'typeof ExpandoExpr3'.
 typeFromPropertyAssignment29.ts(101,22): error TS2339: Property 'prop' does not exist on type 'typeof ExpandoExpr3'.
 typeFromPropertyAssignment29.ts(101,42): error TS2339: Property 'm' does not exist on type 'typeof ExpandoExpr3'.
 
 
-==== typeFromPropertyAssignment29.ts (12 errors) ====
+==== typeFromPropertyAssignment29.ts (18 errors) ====
     function ExpandoDecl(n: number): string {
         return n.toString();
     }
     ExpandoDecl.prop = 2
+    ~~~~~~~~~~~~~~~~
+!!! error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
     ExpandoDecl.m = function(n: number) {
+    ~~~~~~~~~~~~~
+!!! error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
         return n + 1;
     }
     var n: number = ExpandoDecl.prop + ExpandoDecl.m(12) + ExpandoDecl(101).length
     
@@ -130,13 +136,17 @@
         nested.total = n + 1_000_000;
         return nested;
     }
     ExpandoNested.also = -1;
+    ~~~~~~~~~~~~~~~~~~
+!!! error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
     
     function ExpandoMerge(n: number): number {
         return n * 100;
     }
     ExpandoMerge.p1 = 111
+    ~~~~~~~~~~~~~~~
+!!! error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
     namespace ExpandoMerge {
         export var p2 = 222;
     }
     namespace ExpandoMerge {
@@ -146,8 +156,10 @@
     
     namespace Ns {
         function ExpandoNamespace(): void {}
         ExpandoNamespace.p6 = 42;
+        ~~~~~~~~~~~~~~~~~~~
+!!! error TS9023: Assigning properties to functions without declaring them is not supported with --isolatedDeclarations. Add an explicit declaration for the properties assigned to this function.
         export function foo(): typeof ExpandoNamespace {
             return ExpandoNamespace;
         }
     }
@@ -189,8 +201,10 @@
 !!! error TS2339: Property 'm' does not exist on type 'typeof ExpandoClass'.
     
     // Class expressions shouldn't work in typescript either
     var ExpandoExpr3 = class {
+                       ~~~~~
+!!! error TS9022: Inference from class expressions is not supported with --isolatedDeclarations.
         n = 10001;
     }
     ExpandoExpr3.prop = 3
                  ~~~~
